### ISSUE TITLE
"pm2 restart myapp" does not reuse node_args set in ecosystem

### ISSUE THREAD

**Issue by DevSide** [Unitech/pm2#4059]
## What's going wrong?

💚 When I start my app using `pm2 start ecosystem.json --env prod`, everything goes well.

🔴 Then later I use the command `pm2 restart myapp`, the `node_args` doesn't look to be used.

ℹ️ The problem is reproductible on Windows 2008/2012 and MacOS High Sierra

```json5
// ecosystem.json
{
  "apps": [
    {
      "name": "myapp",
      "script": "src/index.js",
      "node_args": "-r esm",
      "instances": 2,
      "exec_mode": "cluster",
      "env": {
        "NODE_ENV": "production"
      },
      "env_prod": {
        "NODE_CONFIG_ENV": "prod"
      }
    }
  ]
}
```

Then after `pm2 restart myapp`

```
$ pm2 info myapp
┌───────────────────┬─────────────────────────────────────────────────┐
│ status            │ errored                                         │
│ name              │ myapp                                           │
│ script path       │ D:\WebApps\myapp\app\src\index.js               │
│ script args       │ N/A                                             │
│ interpreter       │ node                                            │
│ interpreter args  │ N/A                                             │
│ exec mode         │ cluster                                         │
│ node.js version   │ 10.14.1                                         │
│ node env          │ production                                      │
│ watch & reload    │ ?                                               │
│ unstable restarts │ 0                                               │
│ created at        │ N/A                                             │
└───────────────────┴─────────────────────────────────────────────────┘
```

⚠️⚠️ interpreter args is N/A !

```
$ pm2 logs myapp

17|myapp  | (function (exports, require, module, __filename, __dirname) { import { createServer } from '../src/server'
17|myapp  |                                                                      ^
...
```

```
$ pm2 report
===============================================================================
--- PM2 REPORT (Wed Dec 05 2018 17:31:39 GMT+0100 (GMT+01:00)) ----------------
===============================================================================
--- Daemon -------------------------------------------------
pm2d version         : 3.2.2
node version         : 10.14.1
node path            : not found
argv                 : C:\Program Files\nodejs\node.exe,C:\Users\XXXX\AppData\Roaming\npm\node_modules\pm2-windows-service\src\service.js
argv0                : C:\Program Files\nodejs\node.exe
user                 : undefined
uid                  : N/A
gid                  : N/A
uptime               : 1206min
===============================================================================
--- CLI ----------------------------------------------------
local pm2            : 3.2.2
node version         : 10.14.1
node path            : not found
argv                 : C:\Program Files\nodejs\node.exe,C:\Users\XXXX\AppData\Roaming\npm\node_modules\pm2\bin\pm2,report
argv0                : node
user                 : YYYY
===============================================================================
--- System info --------------------------------------------
arch                 : x64
platform             : win32
type                 : Windows_NT
cpus                 : Intel(R) Xeon(R) CPU E5-2695 v2 @ 2.40GHz
cpus nb              : 4
freemem              : 5574324224
totalmem             : 8589398016
home                 : C:\Users\YYYY
===============================================================================
```

**Comment by Noderino** [comment]
Looks like a duplicate of https://github.com/Unitech/pm2/issues/4029

**Comment by DevSide** [comment]
Yeah pretty close issue thanks ! The difference is about using ecosystem config instead of raw parameters when we start the app.

**Comment by Unitech** [comment]
Fixed in pm2 2.3.3:

```bash
npm install pm2@latest -g
pm2 update
```

https://github.com/Unitech/pm2/commit/213f48641beedd1a9b84f74a0f9d963a114eea14
